atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/ConjureClientFactory.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.http.v2;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Proxy;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import com.palantir.atlasdb.config.ServerListConfig;
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.conjure.java.client.config.ClientConfiguration;
+import com.palantir.conjure.java.client.jaxrs.JaxRsClient;
+import com.palantir.conjure.java.okhttp.HostEventsSink;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import com.palantir.tritium.Tritium;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+
+public class ConjureClientFactory {
```

Comment:
would prefer this to not be public otherwise people are gonne start depending on it by accident instead of WC's ConjureClientFactory.

Reply:
Ya it would be nice to keep it package private, but we need to make it visible for a couple packages. I'll rename it `AtlasClientFactory` to avoid any confusion.